pyproject.toml

```diff
@@ -27,6 +27,8 @@ classifiers = [
 ]
 dependencies = [
     'Django>=4.2',
+    'nanoid>=2.0',
+    'selectolax>=0.3',
```

Comment:
Why `nanoid` and `selectolax`?

- Nanoid:

  For many features like CSS scoping, CSS / JS vars, or per-instance JS with `$onLoad`, we need to insert the component's ID into the HTML.

  Previously the component ID was a simply incremental (0, 1, 2, ...). But making this compatible with HTML fragments makes the incremental IDs not suitable.

  Because I could render a full HTML document with components with IDs 0, 1, 2, 3, 4. Then, for whatever reason, the Django server crashes and restarts. And only then user makes an action that triggers a rendering of an HTML fragment.

  In the case if the IDs were incremental, the fragment's HTML could contain component IDs e.g. 0 and 1. But components with IDs 0 and 1 are already in the HTML from the initial render.

  So, instead, we need to use global / universal IDs, so the IDs are not dependent on the server's state like incrementing IDs. And for that I used nanoid.

  As described in a comment here https://github.com/EmilStenstrom/django-components/pull/688/files#diff-b07310aac6ba2705304fd226eda2e8ca6ac5d0324b2d84d6ba83b47f06744169R14, just 6 digit-long ID already has a chance of collision 1 in 3M

- Seletolax:

  For the features above, we need to insert various marks into the generated HTML. E.g. to mark each component's HTML with the respective IDs.

  For the scoped CSS, we decided to implement it like Vue did, by marking all HTML nodes with the component's ID. For something like that, regex is not sufficient, and we need to properly parse the HTML.

  There are multiple libraries for that, but it seems that selectolax is by far the faster ([See comparison](https://medium.com/@yahyamrafe202/4f268ddea8df))

Reply:
I managed to remove nanoid! It was actually really small library, so I copied to code in https://github.com/EmilStenstrom/django-components/pull/688#discussion_r1785985901.

As for why I didn't just use `random.random()` or similar - I don't know enough about cryptography, so I prefer to choose a solution that claims to have what I'm looking for. And nanoid claims to have low chance of collision and optimized for small IDs, which is the case here.

I explored a bit here https://chatgpt.com/share/670512bb-e4b8-8004-9df5-0c9dc60762fb